### ISSUE TITLE
Paste directly from $CM_LAUNCHER

### ISSUE THREAD

**Issue by stef204** [cdown/clipmenu#129]
Clipmenu 6.1.0

When I bring up the Clipmenu list of copied items in Rofi, select one item and the press `enter` nothing happens.. This is a giant PITA (to me.)

`Crtl+V` and `Shift-Insert` both work but that is not how I want to use it.
I should be able to select an item on the list and just hit `enter` for it to be pasted wherever cursor is (or "current window".)

What am I doing wrong?

**Comment by cdown** [comment]
That's simply not how clipmenu works, or has ever worked :-)

But other than that, this is a dupe of #125

**Comment by cdown** [comment]
Oh wait, #125 is misleadingly titled, so let's keep this separate.

But yeah, this has never been a feature clipmenu offered. Patches are welcome, though :-)

**Comment by stef204** [comment]
I don't understand why you think if should be "off" by default.  How can one beat the ease of use and flow of 1) selecting an item and 2) pasting an item by just using `enter`?

That is how it should work.

How can I make it work in this fashion currently? `&& xdotool key shift+Insert` does not seem to work for me.

And I had missed #125 (as you mentioned, title did not bring focus to the issue but it is indeed the same problem , it seems.)

Using i3 if that helps,

**Comment by cdown** [comment]
> I don't understand why you think if should be "off" by default. How can one beat the ease of use and flow of 1) selecting an item and 2) pasting an item by just using enter?
>
> That is how it should work.

Your assertiveness that that is how it *should* work is really not matched by the experience of the majority of clipmenu users.

Personally I do not ever want a paste to happen without my explicit say so, since there are many reasons focus could change and have the output pasted to the wrong window (dialog windows, X11 urgent hint, etc...). If that's something sensitive, that's a disaster, and there's no way we're going to have that on by default.

**Comment by stef204** [comment]
I totally see your point and agree with it.  It's very sensitive and you could end up pasting your password on a webpage if focus changes to it from another window you wanted to paste into.

But is that not managed by just being focused on the window you intend to paste in in the first place?

I'll try it this way and if I run into trouble will remove the `xdotool`.

To stay on the practical side, how does one achieve the 'select and paste in one shot' (by pressing `enter`) other than using the xdotool trick?

Or that's just not a feature at all and no plans to make it optional?

**Comment by cdown** [comment]
> But is that not managed by just being focused on the window you intend to paste in in the first place?

No, because typically `$CM_LAUNCHER` grabs focus with `XSetInputFocus`, and it's up to the window manager how it puts it back, whereas this doesn't happen with the "normal" paste case.

> To stay on the practical side, how does one achieve the 'select and paste in one shot' (by pressing enter) other than using the xdotool trick?
>
> Or that's just not a feature at all and no plans to make it optional?

To quote from my reply above: patches are welcome. :-)

**Comment by mi-lad** [comment]
@stef204 I have created PR #137 to do the paste via `xdotool`

**Comment by wwalker** [comment]
I wrote my own tool (2 lines) to do what @stef204 is asking for.  Because not all tools accept Cntl-V (or Shift-Insert, or middle mouse click), I mas my tool type the output instead of pasting the output.

I thought it was great, until I selected a 50 line selection.  It started `type`ing and I looked at a different window, and moved focus there ....   and part of the 60 line selection was run at the terminal!!  Fortunately, there was no dangerous commands in the selection, but there could have been.

I'm also often bit by some popup stealing focus.  But having said that I'll try to provide a patch (off be default).
